src/internal/components/pluggableVisualizations/geoChart/PluggableGeoPushpinChart.tsx

```diff
@@ -0,0 +1,98 @@
+// (C) 2019-2020 GoodData Corporation
+import * as React from "react";
+import { render } from "react-dom";
+
+import cloneDeep = require("lodash/cloneDeep");
+import get = require("lodash/get");
+import set = require("lodash/set");
+
+import * as BucketNames from "../../../../constants/bucketNames";
+import {
+    IReferencePoint,
+    IExtendedReferencePoint,
+    IVisConstruct,
+    IVisualizationProperties,
+} from "../../../interfaces/Visualization";
+import { configurePercent, configureOverTimeComparison } from "../../../utils/bucketConfig";
+import { PluggableBaseChart } from "../baseChart/PluggableBaseChart";
+import { BUCKETS, METRIC, ATTRIBUTE, GEO_ATTRIBUTE } from "../../../constants/bucket";
+import { GEO_PUSHPIN_CHART_UICONFIG } from "../../../constants/uiConfig";
+import { sanitizeFilters, getStackItems, getBucketItemsByType } from "../../../utils/bucketHelper";
+import { setGeoPushPinUiConfig } from "../../../utils/uiConfigHelpers/geoPushpinChartUiConfigHelper";
+import { removeSort } from "../../../utils/sort";
+import { getReferencePointWithSupportedProperties } from "../../../utils/propertiesHelper";
+import { VisualizationTypes } from "../../../../constants/visualizationTypes";
+import UnsupportedConfigurationPanel from "../../configurationPanels/UnsupportedConfigurationPanel";
+
+export class PluggableGeoPushpinChart extends PluggableBaseChart {
+    constructor(props: IVisConstruct) {
+        super(props);
+        this.type = VisualizationTypes.PUSHPIN;
+        this.supportedPropertiesList = [];
+        this.initializeProperties(props.visualizationProperties);
+    }
+
+    public getExtendedReferencePoint(referencePoint: IReferencePoint): Promise<IExtendedReferencePoint> {
+        const clonedReferencePoint = cloneDeep(referencePoint);
+        let newReferencePoint: IExtendedReferencePoint = {
+            ...clonedReferencePoint,
+            uiConfig: cloneDeep(GEO_PUSHPIN_CHART_UICONFIG),
+        };
+
+        const buckets = get(clonedReferencePoint, BUCKETS, []);
+        const locations = getBucketItemsByType(buckets, BucketNames.LOCATION, [GEO_ATTRIBUTE]);
+        const segments = getStackItems(buckets, [GEO_ATTRIBUTE, ATTRIBUTE]);
+        const measuresSize = getBucketItemsByType(buckets, BucketNames.SIZE, [METRIC]);
+        const measuresColor = getBucketItemsByType(buckets, BucketNames.COLOR, [METRIC]);
+
+        set(newReferencePoint, BUCKETS, [
+            {
+                localIdentifier: BucketNames.LOCATION,
+                items: locations,
+            },
+            {
+                localIdentifier: BucketNames.SIZE,
+                items: measuresSize,
+            },
+            {
+                localIdentifier: BucketNames.COLOR,
+                items: measuresColor,
+            },
+            {
+                localIdentifier: BucketNames.SEGMENT,
```

Comment:
Please use new `export const SEGMENT_BY = "segmentBy";`
It already used here in SDK https://github.com/gooddata/gooddata-react-components/pull/1312/files#diff-4ca1203f42326c75f48bd980cf10ca37R34